File: 백준/Silver/26069. 붙임성 좋은 총총이/붙임성 좋은 총총이.swift
// [Int] N (immutable): 사람들이 만난 기록의 수 (범위(자연수): 1 <= N <= 1,000)
let N: Int = Int(readLine() ?? "1") ?? 1

// [Set<String>] rainbow (mutable): 무지개 댄스를 추는 사람의 집합
// 초기값: ChongChong (처음부터 무지개 댄스를 추는 사람)
var rainbow: Set<String> = ["ChongChong"]


for _ in 0..<N {
    // [Array<String>] log (immutable): 만난 두 사람을 저장한 리스트 (길이: 2)
    let log: [String] = (readLine() ?? "a b")
        .split(separator: " ")
        .map(String.init)
    
    // 만약 log 속 사람 중에 한 명이라도 rainbow 집합에 포함되어 있다면
    // 두 사람을 모두 rainbow 집합에 추가함.
    for people in log {
        if (rainbow.contains(people)) {
            rainbow.formUnion(log)
            break
        }
    }
}

// 무지개 댄스를 추는 사람의 수를 출력
print(rainbow.count)

File: 백준/Silver/26069. 붙임성 좋은 총총이/붙임성 좋은 총총이.py
# [int] N: 사람들이 만난 기록의 수 (범위(자연수): 1 <= N <= 1,000)
N = int(input())

# [set(str)] rainbow: 무지개 댄스를 추는 사람의 집합
rainbow = set()
# ChongChong은 처음부터 무지개 댄스를 춘다.
rainbow.add("ChongChong")

for _ in range(N):
    # [list(str)] log: 만난 두 사람을 저장한 리스트 (길이: 2)
    log = input().split()
    
    # 만약 log 속 사람 중에 한 명이라도 rainbow 집합에 포함되어 있다면
    # 두 사람을 모두 rainbow 집합에 추가함.
    for people in log:
        if people in rainbow:
            rainbow.update(log)
            break

# 무지개 댄스를 추는 사람의 수를 출력
print(len(rainbow))